lib/vanagon/driver.rb

```diff
@@ -76,15 +76,16 @@ def run
       @engine.startup(@workdir)
 
       puts "Target is #{@engine.target}"
-
-      install_build_dependencies
-      @project.fetch_sources(@workdir)
-      @project.make_makefile(@workdir)
-      @project.make_bill_of_materials(@workdir)
-      @project.generate_packaging_artifacts(@workdir)
-      @engine.ship_workdir(@workdir)
-      @engine.dispatch("(cd #{@engine.remote_workdir}; #{@platform.make})")
-      @engine.retrieve_built_artifact
+      Vanagon::Utilities.retry_with_timeout(3, 3600) do
+        install_build_dependencies
+        @project.fetch_sources(@workdir)
+        @project.make_makefile(@workdir)
+        @project.make_bill_of_materials(@workdir)
+        @project.generate_packaging_artifacts(@workdir)
+        @engine.ship_workdir(@workdir)
+        @engine.dispatch("(cd #{@engine.remote_workdir}; #{@platform.make})")
+        @engine.retrieve_built_artifact
+      end
```

Comment:
should each of these things be in a retry block? Is it just going to try everything all over again if there's a failure? Like, if the `dispatch` step fails, are we going to re-fetch all the sources?